pmm/pmm2-upgrade-portal-tests.groovy

```diff
@@ -0,0 +1,267 @@
+library changelog: false, identifier: 'lib@master', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/Percona-Lab/jenkins-pipelines.git'
+]) _
+
+void runStagingServer(String DOCKER_VERSION, CLIENT_VERSION, CLIENT_INSTANCE, SERVER_IP, ADMIN_PASSWORD = "admin") {
+    stagingJob = build job: 'aws-staging-start', parameters: [
+        string(name: 'DOCKER_VERSION', value: DOCKER_VERSION),
+        string(name: 'CLIENT_VERSION', value: CLIENT_VERSION),
+        string(name: 'CLIENTS', value: ""),
+        string(name: 'CLIENT_INSTANCE', value: CLIENT_INSTANCE),
+        string(name: 'DOCKER_ENV_VARIABLE', value: '-e PMM_DEBUG=1 -e PERCONA_TEST_SAAS_HOST=check-dev.percona.com -e PERCONA_TEST_CHECKS_PUBLIC_KEY=RWTg+ZmCCjt7O8eWeAmTLAqW+1ozUbpRSKSwNTmO+exlS5KEIPYWuYdX -e PERCONA_TEST_CHECKS_INTERVAL=10s'),
+        string(name: 'SERVER_IP', value: SERVER_IP),
+        string(name: 'NOTIFY', value: 'false'),
+        string(name: 'DAYS', value: '1'),
+        string(name: 'ADMIN_PASSWORD', value: ADMIN_PASSWORD)
+    ]
+    env.VM_IP = stagingJob.buildVariables.IP
+    env.VM_NAME = stagingJob.buildVariables.VM_NAME
+    def clientInstance = "yes";
+    if ( CLIENT_INSTANCE == clientInstance ) {
+        env.PMM_URL = "http://admin:${ADMIN_PASSWORD}@${SERVER_IP}"
+        env.PMM_UI_URL = "http://${SERVER_IP}/"
+    }
+    else
+    {
+        env.PMM_URL = "http://admin:${ADMIN_PASSWORD}@${VM_IP}"
+        env.PMM_UI_URL = "http://${VM_IP}/"
+    }
+}
+
+void destroyStaging(IP) {
+    build job: 'aws-staging-stop', parameters: [
+        string(name: 'VM', value: IP),
+    ]
+}
+
+pipeline {
+    agent {
+        label 'agent-amd64'
+    }
+    environment {
+        PORTAL_USER_EMAIL=credentials('PORTAL_USER_EMAIL')
+        PORTAL_USER_PASSWORD=credentials('PORTAL_USER_PASSWORD')
+        OKTA_TOKEN=credentials('OKTA_TOKEN')
+        SERVICENOW_LOGIN=credentials('SERVICENOW_LOGIN')
+        SERVICENOW_PASSWORD=credentials('SERVICENOW_PASSWORD')
+        SERVICENOW_DEV_URL=credentials('SERVICENOW_DEV_URL')
+        OAUTH_DEV_CLIENT_ID=credentials('OAUTH_DEV_CLIENT_ID')
+        PORTAL_BASE_URL=credentials('PORTAL_BASE_URL')
+    }
+    parameters {
+        string(
+            defaultValue: 'main',
+            description: 'Tag/Branch for pmm-ui-tests repository',
+            name: 'GIT_BRANCH')
+        string(
+            defaultValue: '',
+            description: 'Commit hash for the branch',
+            name: 'GIT_COMMIT_HASH')
+        string(
+            defaultValue: 'perconalab/pmm-server:2.27.0',
```

Comment:
thats a wrong tag, we don't have perconalab/pmm-server:2.27.0

I would recommend use `choices: ['2.27.0', '2.26.0', '2.25.0'],` kind of setup for server, ideally upgrade jobs are run from already released version to dev-latest hence `percona/pmm-server` is already fixed. 

See how we do this in upgrade job

Reply:
https://hub.docker.com/r/perconalab/pmm-server/tags?page=1&name=2.27 - There is a 2.27.0 tag for perconalb/pmm-server.

Okay I will add choices, but as @nikita-b requested will keep switch to select between perconalab/pmm-sever and percona/pmm-server

in pmm2-upgrade job we have just a string.

Reply:
in that case I probably got it confused with matrix job, in that case just use the most recently released version label, which should be `percona/pmm-server:2`